src/components/Navigator/NavigatorDataProvider.vue

```diff
@@ -49,16 +52,26 @@ export default {
       const matches = /(\/documentation\/(?:[^/]+))\/?/.exec(technology.url);
       return matches ? matches[1] : '';
     },
+    versionedIndexData() {
+      return patchToVersion(DocumentationTopicStore.state.preferredVersion,
+        this.allIndexData);
+    },
+    versionedNavigationIndex() {
+      const { interfaceLanguages } = this.versionedIndexData || {};
+      const preFreezedNav = interfaceLanguages || this.navigationIndex;
+      return Object.freeze(preFreezedNav);
```

Comment:
You no longer use `navigationIndex`, so you may remove it entirely.

Also there is no point in freezing inside the `computed`, so we can just return the `interfaceLanguages`. I added a fallback just in case.

```suggestion
      const { interfaceLanguages } = this.versionedIndexData || {};
      return interfaceLanguages || { [Language.swift.key.url]: [] }
```

Reply:
Thanks @dobromir-hristov for the object freeze feedback. for the fallback proposed, do we need it at all? we didn't have it before. Just wondering

Reply:
@dobromir-hristov Not sure how to fix this correctly, kept it this way for now in the most recent PR, but needs to be updated to reflect this feedback.